Result/Result.swift

```diff
@@ -1,10 +1,35 @@
 //  Copyright (c) 2015 Rob Rix. All rights reserved.
 
+#if swift(>=4.2)
+#if compiler(>=5)
```

Comment:
I don't think this maintains the requirements for semantic version compatibility, so I think we need to drop support for Swift 4 to do this. (And we should do this!)

If we conditionally define the enum, then the output is different if Result is compiled in Swift 4 mode or Swift 5 mode. Since semantic versioning was defined by TPW and he writes Ruby, it's not totally clear whether it's meant to be source- or binary-compatible, but I think the safe thing is to require both.

Reply:
>  I think we need to drop support for Swift 4 to do this. (And we should do this!)

Just changing the major version is sufficient, and it seems there is no need to make it impossible to build with older Swift compiler.🤔 

> If we conditionally define the enum, then the output is different if Result is compiled in Swift 4 mode or Swift 5 mode.

It is not affected by the `-swift-version` flag. When using the Swift 5+ compiler, `Swift.Result` will be used.

Reply:
Sorry, edited previous comment. 🙇

Reply:
Oh, I see! Tricky!